### PR TITLE
D8CORE-1279: Fix for subtitle/subhead styles

### PR DESCRIPTION
**Pull request by cjwest**
# Maybe READY FOR REVIEW - advice needed
- (Edit the above to reflect status)

# Summary
- TL;DR - Fixes the styles for the subtitle/subhead

# Needed By (Date)
- Before sprint end

# Urgency
- Would be needed for any site using these styles

# Steps to Test

1. Check out this branch
1. Clear Cache
1. In the editor add "Subtitle" to text
1. Save.
1. Verify that the Subtitle text is:

Screen Shot 2020-02-19 at 6 01 37 PM

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

D8Core

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/D8CORE-1279
- Other PRs: This is taking over these PRs:
https://github.com/SU-SWS/stanford_profile/pull/109
https://github.com/SU-SWS/stanford_profile/pull/107

- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)

Please see https://stanfordits.atlassian.net/browse/D8CORE-1279?focusedCommentId=90556&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-90556 

- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
comments left in Jira ticket.

**Comment by cjwest** [comment]
See [comment in ticket](https://stanfordits.atlassian.net/browse/D8CORE-1279?focusedCommentId=90674&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-90674).

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg.scss [hunk 1]
```diff
@@ -59,136 +59,150 @@
...
+        text-align: center;
+      }
+    }
 
-    &:first-child {
-      // Remove margin-top and padding-top when it is the first item on the page
-      @include margin(0 null null null);
-      @include padding(0 null null null);
+    // Buttons.
+    @at-root .su-button,
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg.scss]: `@at-root` is not what we need here as it refers to `body`

This statement outputs the following selector:
```
body .su-button,
.su-wysiwyg-text .su-button {
...
}
```

We want to output:
```
.su-wysiwyg-text .su-button,
.su-wysiwyg-text p.su-button {
...
}
```

**Comment by sherakama** [comment]
@cjwest please have a review of the changes. I took the liberty to split up the large wysiwyg.scss file into multiple partials as it was getting scattered and similar selectors were in different parts of the document. Hopefully, it will help organize future overrides and edits.

You were close with your approach but were a little heavy handed. Not everything needed to be nested in the `p` tag as not everything is applied to a p tag (like buttons) and not everything was a override of a css selector out of Decanter. I refactored so that the p tag and the classes stand on their own again and only doubled down on the items that were overrides of css class styles from Decanter.

Also, the `@at-root` option was a great choice but missed the mark ever so slightly. Have a read of: https://stackoverflow.com/questions/17268051/sass-combining-parent-using-ampersand-with-type-selectors

In that article it explains that the `@at-root` selector brings you back to the FIRST selector in the tree. In this case it was .su-wysiwyg-text or body depending on where it was being used. You can pair the `@at-root` selector with the `selector-unify` sass function to combine your current selector tree to get the output you were looking for. Using that approach and the `&` as the parent selector you can combine the two to look like the following.

```
// Subtitle styles.
  .su-subheading {
    @at-root #{selector-unify(&, p)},
    & {
      @include subheading;
    }
  }
```

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      text-align: center;
+    }
+  }
+
+  // Big bold beautiful callout text.
+  .su-callout-text {
+    @include margin(0 auto 1em 0);
+    @include modular-spacing('margin', -3 null 1 null);
+    display: block;
+    font-size: 28px;
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: This should be `2.8em`

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    font-style: normal;
+    font-weight: bold;
+    line-height: 33px;
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: This should be ems as well. `3.3em`

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  // Caption Styles.
+  .caption {
+    figcaption {
+      text-align: center;
+    }
+  }
+
+  // Big bold beautiful callout text.
+  .su-callout-text {
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]: This selector probably needs the 
```
@at-root #{selector-unify(&, p)},
    & {
```
approach as this selector will lose out to the `p` selector on the font size.

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    @include margin(0 auto 1em 0);
+    @include modular-spacing('margin', -3 null 1 null);
+    display: block;
+    font-size: 28px;
+    font-style: normal;
+    font-weight: bold;
```
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]:
```suggestion
    font-weight: $su-font-bold;
```

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss (hunk 2 above)
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]:
```suggestion
    font-size: 2.8em;
```

File: modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss (hunk 3 above)
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/paragraph-types/wysiwyg/_paragraph.scss]:
```suggestion
    line-height: 3.3rem;
```

**Comment by cjwest** [comment]
Wow @sherakama, you've been busy. Thank you for fixing this all up. 
and thanks for demonstrating how to use {{selector-unify}}. I think I have some reading to do on SASS!
The styles are rendering as expected in my examples. However, it appears I cannot approve and merge.
